onnxruntime/core/providers/webnn/builders/impl/normalization_op_builder.cc

```diff
@@ -116,15 +112,34 @@ Status NormalizationOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder
   options.set("epsilon", helper.Get("epsilon", 1e-05f));
 
   emscripten::val output = emscripten::val::undefined();
-  if (op_type == "LayerNormalization") {
+  if (op_type == "BatchNormalization") {
+    ORT_RETURN_IF_NOT(input_defs.size() == 5, "BatchNormalization requires five inputs.");
+    emscripten::val mean = model_builder.GetOperand(input_defs[3]->Name());
+    emscripten::val variance = model_builder.GetOperand(input_defs[4]->Name());
+    // Enlarge 1-D mean and variance to new scale shape.
+    emscripten::val reshape_mean =
+        model_builder.GetBuilder().call<emscripten::val>("reshape", mean, emscripten::val::array(new_scale_shape));
+    emscripten::val reshape_variance =
+        model_builder.GetBuilder().call<emscripten::val>("reshape", variance, emscripten::val::array(new_scale_shape));
+
+    std::vector<size_t> axes = {0};
+    for (size_t i = 2; i < rank; i++) {
+      axes.push_back(i);
+    }
+
+    options.set("axes", emscripten::val::array(axes));
+    options.set("mean", reshape_mean);
+    options.set("variance", reshape_variance);
+    output = model_builder.GetBuilder().call<emscripten::val>("meanVarianceNormalization", input, options);
+  } else if (op_type == "LayerNormalization") {
     int64_t axis = helper.Get("axis", -1);
     axis = HandleNegativeAxis(axis, rank);
-    std::vector<int32_t> axes(rank - narrow<size_t>(axis));
+    std::vector<size_t> axes(rank - narrow<size_t>(axis));
     std::iota(axes.begin(), axes.end(), axis);
     options.set("axes", emscripten::val::array(axes));
     output = model_builder.GetBuilder().call<emscripten::val>("meanVarianceNormalization", input, options);
   } else if (op_type == "InstanceNormalization") {
-    std::vector<int32_t> axes;
+    std::vector<size_t> axes;
```

Comment:
axes was changed from `int32_t` -> `size_t`, but doesn't the IDL require `uint32_t`?

```
dictionary MLMeanVarianceNormalizationOptions {
  MLOperand mean; // optional
  MLOperand variance; // optional
  MLOperand scale;
  MLOperand bias;
  float epsilon = 1e-5;
  sequence<[EnforceRange] unsigned long> axes; // <-----------
};
```

Reply:
Ah, good point, I will fix that.